tezos/new_context/src/working_tree/working_tree.rs

```diff
@@ -1004,7 +999,7 @@ impl WorkingTree {
                             None => Ok(()),
                             Some(entry) => self.get_entries_recursively(
                                 entry,
-                                child_node.entry_hash.borrow().clone(),
+                                &child_node.entry_hash.borrow().clone().unwrap(),
```

Comment:
What guarantees that this entry hash will be present? why not `child_node.entry_hash()?` like above?

Reply:
The other thing is that it is being cloned, to be passed as a reference, that will then be cloned again by the inner call.

The `commit_hash` recalculation is fine, I am just not entirely sure about this part here.

Reply:
@tizoc The hashes for the whole tree are actually already computed, before the call to `get_entries_recursively` here:
https://github.com/tezedge/tezedge/blob/protocol-runner-storage/tezos/new_context/src/working_tree/working_tree.rs#L738

I will fix the double clone

Reply:
Ahhh, good point.